File: app/src/main/java/com/princeparadoxes/watertracker/ui/screen/main/statistic/StatisticChartView.java
package com.princeparadoxes.watertracker.ui.screen.main.statistic;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.util.AttributeSet;
import android.view.LayoutInflater;
import android.widget.LinearLayout;

import com.github.mikephil.charting.charts.LineChart;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.data.LineData;
import com.github.mikephil.charting.data.LineDataSet;
import com.princeparadoxes.watertracker.R;

import java.util.ArrayList;

import butterknife.BindColor;
import butterknife.BindView;
import butterknife.ButterKnife;

public class StatisticChartView extends LinearLayout {

    @BindView(R.id.statistic_cart_view_chart)
    LineChart mChartView;

    @BindColor(R.color.accent)
    int mAccentColor;
    @BindColor(R.color.accent_alpha)
    int mAccentAlphaColor;

    public StatisticChartView(Context context, AttributeSet attrs) {
        super(context, attrs);
        LayoutInflater.from(context).inflate(R.layout.statistic_chart_view, this);
        ButterKnife.bind(this);
        // add a selection listener
        mChartView.setDescription(null);
        // Hide the legend
        mChartView.getLegend().setEnabled(false);
        mChartView.getAxisLeft().setDrawGridLines(false);
        mChartView.getAxisRight().setDrawGridLines(false);
        mChartView.getXAxis().setDrawGridLines(false);

    }

    public void bindView(@NonNull float[] xValues, @NonNull float[] yValues) {
        if (xValues.length != yValues.length) {
            throw new IllegalArgumentException("xValues length must be equals yValues length");
        }
        ArrayList<Entry> entries = new ArrayList<>();
        for (int i = 0; i < xValues.length; i++) {
            entries.add(new Entry(xValues[i], yValues[i]));
        }
        LineDataSet dataSet = new LineDataSet(entries, "");
        dataSet.setDrawIcons(false);
        dataSet.setValueTextColor(Color.WHITE);
        dataSet.setColor(Color.WHITE);
        dataSet.setCircleColor(Color.WHITE);

        LineData data = new LineData(dataSet);
        mChartView.setData(data);
        mChartView.invalidate();
    }

}

File: app/src/main/java/com/princeparadoxes/watertracker/presentation/screen/statistic/StatisticChartView.kt
package com.princeparadoxes.watertracker.presentation.screen.statistic

import android.content.Context
import android.graphics.Color
import android.support.v4.content.ContextCompat
import android.util.AttributeSet
import android.view.LayoutInflater
import android.widget.LinearLayout
import com.github.mikephil.charting.animation.Easing
import com.github.mikephil.charting.charts.BarChart
import com.github.mikephil.charting.components.XAxis
import com.github.mikephil.charting.data.BarData
import com.github.mikephil.charting.data.BarDataSet
import com.github.mikephil.charting.data.BarEntry
import com.princeparadoxes.watertracker.R

class StatisticChartView(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {

    private val chartView by lazy { findViewById(R.id.statistic_cart_view_chart) as BarChart }

    private val accentColor by lazy { ContextCompat.getColor(context, R.color.accent) }
    private val accentAlphaColor by lazy { ContextCompat.getColor(context, R.color.accent_alpha) }

    init {
        LayoutInflater.from(context).inflate(R.layout.statistic_chart_view, this)
        chartView.apply {
            description = null
            legend.isEnabled = false
            isDoubleTapToZoomEnabled = false
        }
        chartView.xAxis.apply {
            setDrawGridLines(false)
            textColor = Color.TRANSPARENT
            axisLineColor = Color.WHITE
            position = XAxis.XAxisPosition.BOTTOM
        }
        chartView.axisLeft.apply {
            setDrawGridLines(false)
            textColor = Color.WHITE
            axisMaximum = 2200F
            axisMinimum = 0F
            axisLineColor = Color.WHITE
        }
        chartView.axisRight.apply {
            setDrawGridLines(false)
            textColor = Color.TRANSPARENT
            axisMaximum = 2200F
            axisMinimum = 0F
            axisLineColor = Color.TRANSPARENT
        }
    }

    fun bindView(xValues: FloatArray, yValues: FloatArray) {
        if (xValues.size != yValues.size) {
            throw IllegalArgumentException("xValues length must be equals yValues length")
        }
        val entries = xValues.indices.map { BarEntry(xValues[it], yValues[it]) }
        val dataSet = BarDataSet(entries, "").apply {
            setDrawIcons(false)
            valueTextColor = Color.TRANSPARENT
            color = Color.WHITE
        }

        val data = BarData(dataSet)
        data.isHighlightEnabled = false
        chartView.data = data
        chartView.invalidate()
        chartView.animateY(300)
    }

}
